### ISSUE TITLE
Running KDiskMark on a Raspberry Pi

### ISSUE THREAD

**Issue by raspberrypieman** [JonMagon/KDiskMark#156]
I have been using CrystalDiskMark on Windows for many years and love it. KDiskMark seems to be an ideal equivalent on Linux. However, I want to run it on a Raspberry Pi4b running Raspbian Bookworm.  I have no idea about compiling it from source, I just want a installable binary. I found something in this repository about using PPA but it would not work on my system. I think my fellow Raspberry Pi users would love to use this application but are put off because they cannot simply use apt install ...
Can anyone help, please?

**Comment by raspberrypieman** [comment]
Thank-you for your speedy response.

Sorry, for my lack of Linux knowledge.
I have downloaded the file: kdiskmark_3.1.4-bionic-2_arm64.deb
to my Raspberry Pi4b, but do not know what to do next?

Martyn

**Comment by raspberrypieman** [comment]
Excellent,
     It installed OK, but I was not able to run on my Pi4 as it is
currently running OMV and only has a command line interface - will try
later with another os.
However I did install it on a Raspberry Pi5 and it worked like a dream!
Although it ran ok, it did report the following line 5 times: libpng
warning iCCP: known incorrect sRGB profile.

Thanks very much for your help and patience,
Martyn

**Comment by EnTerr** [comment]
```
@rpi:~ $ sudo apt install ./kdiskmark_3.1.4-bionic-2_arm64.deb 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'kdiskmark:arm64' instead of './kdiskmark_3.1.4-bionic-2_arm64.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 kdiskmark:arm64 : Depends: fio:arm64 (>= 3.1) but it is not installable
                   Depends: libc6:arm64 (>= 2.17) but it is not installable
                   Depends: libgcc1:arm64 (>= 1:3.0) but it is not installable
                   Depends: libpolkit-qt5-1-1:arm64 (>= 0.112.0) but it is not installable
                   Depends: libqt5core5a:arm64 (>= 5.5.0) but it is not installable
                   Depends: libqt5dbus5:arm64 (>= 5.0.2) but it is not installable
                   Depends: libqt5gui5:arm64 (>= 5.7.0) but it is not installable
                   Depends: libqt5network5:arm64 (>= 5.0.2) but it is not installable
                   Depends: libqt5widgets5:arm64 (>= 5.2.0~alpha1) but it is not installable
                   Depends: libstdc++6:arm64 (>= 5) but it is not installable
E: Unable to correct problems, you have held broken packages.
```
i guess i could have guessed by the name it's 64-bit version w/o trying 

Please compile a 32-bit version?

Use case, Raspberry Pi Zero 2 (a miniaturised RPi 3, the latest and greatest in small-factor RPi) has only 0.5GB RAM and makes no sense to install 64-bit OS, while still very important to check the speeds of microSD card from which boots. There's the builtin "Agnostics" (`sh /ush/share/agnostics/sdtest.sh` for cmd line) which tests 3 values (sequential write MB/s, random write and read IOPS) but reports are somewhat unstable, can vary wildly since does not do repeat trials to report median/mean. 

I'd argue if compiling only one version, it should be 32-bit, since can run on 64-bit OS as well, when the reverse is not true. And SD cards are way, way, way slower for 32-vs-64 bit to make a measurable difference.

**Comment by JonMagon** [comment]
There is an armhf package: https://launchpad.net/~jonmagon/+archive/ubuntu/kdiskmark/+build/26322422/+files/kdiskmark_3.1.4-bionic-2_armhf.deb

**Comment by EnTerr** [comment]
Yes! Thank you so very much @JonMagon - i could not figure out before how to browse launchpad.net looking for releases. I am glad to report that the arm-hf package works on RPiZ2 + 32bit Raspbian 11/Bullseye
